tests/__init__.py

```diff
@@ -27,15 +27,15 @@ def compare_item_with_dict(item: Item, data: dict):
     """
     for key, value in data.items():
         if key == 'additionalProperty':
-            value = AdditionalProperty.from_list(value)
+            value = [AdditionalProperty.from_dict(v) for v in value]
```

Comment:
Generic tests can be developed for the new method, e.g.
* test nested attrs classes
* test nested lists with attr classes
* test nested dicts with attr classes
* test plain lists of basic types
* etc

With these tests in place, it won't be required to perform individual nesting tests to each new defined Item.